[DOCS] Update of Sample Articles for master (#20957)
openvinotoolkit/openvino: docs/articles_en/learn_openvino/openvino_samples/get_started_demos.md

Once the prerequisites have been installed, perform the following steps:

1. :ref:`Build Samples <build-samples>`.
2. :ref:`Select a Sample <select-sample>`.
3. :ref:`Download a suitable model <download-model>`.
4. :ref:`Download media files used as input, if necessary <download-media>`.

Once you perform all the steps, you can :ref:`run inference with the chosen sample application <run-inference>` to see the results. 

.. _build-samples:

Build the Sample Applications
================================

Select a sample you want to use from the :doc:`OpenVINO Samples <openvino_docs_OV_UG_Samples_Overview>` page, and follow the instructions below to build it in your operating system.

.. note::

   Some samples may also require `OpenCV <https://github.com/opencv/opencv/wiki/BuildOpenCV4OpenVINO>`__ to run properly. Make sure to install it for use with vision-oriented samples. 

Instructions below show how to build sample applications with CMake. If you are interested in building them from source, check the `build instructions on GitHub <https://github.com/openvinotoolkit/openvino/blob/master/docs/dev/build.md>`__ .

.. tab-set::

   .. tab-item:: Linux
      :sync: linux

>       The officially supported Linux build environment is the following:
> 
>       * Ubuntu 18.04 LTS 64-bit or Ubuntu 20.04 LTS 64-bit
>       * GCC 7.5.0 (for Ubuntu 18.04) or GCC 9.3.0 (for Ubuntu 20.04)
>       * CMake version 3.10 or higher

Reviewer: ```suggestion
      Before you build the samples on Linux, refer to the the :doc:`system requirements <system_requirements>` page.
      
```

                  mkdir build
               
               .. note:: 
               
                  If you ran the Image Classification verification script during the installation, the C++ samples build directory is created in your home directory: ``~/openvino_cpp_samples_build/``
               
            2. Go to the created directory:
               
               .. code-block:: sh
                  
                  cd build
               
            3. Run CMake to generate the Make files for release configuration. For example, for C++ samples:
            
                .. code-block:: sh
                  
                   cmake -DCMAKE_BUILD_TYPE=Release <INSTALL_DIR>/samples/cpp
               
            
            4. Run ``make`` to build the samples:
            
               .. code-block:: sh
                  
                  cmake --build . --parallel
            
            For the release configuration, the sample application binaries are in ``<path_to_build_directory>/<architecture>/Release/``;
            for the debug configuration — in ``<path_to_build_directory>/<architecture>/Debug/``.

   .. tab-item:: Windows
      :sync: windows

>       The recommended Windows build environment is the following:
> 
>       * Microsoft Windows 10
>       * Microsoft Visual Studio 2019
>       * CMake version 3.10 or higher

Reviewer: ```suggestion
      Before you build the samples on Windows, refer to the the :doc:`system requirements <system_requirements>` page.

```

Once the prerequisites have been installed, perform the following steps:

1. :ref:`Build Samples <build-samples>`.
2. :ref:`Select a Sample <select-sample>`.
3. :ref:`Download a suitable model <download-model>`.
4. :ref:`Download media files used as input, if necessary <download-media>`.

Once you perform all the steps, you can :ref:`run inference with the chosen sample application <run-inference>` to see the results. 

.. _build-samples:

Build the Sample Applications
================================

Select a sample you want to use from the :doc:`OpenVINO Samples <openvino_docs_OV_UG_Samples_Overview>` page, and follow the instructions below to build it in your operating system.

.. note::

   Some samples may also require `OpenCV <https://github.com/opencv/opencv/wiki/BuildOpenCV4OpenVINO>`__ to run properly. Make sure to install it for use with vision-oriented samples. 

Instructions below show how to build sample applications with CMake. If you are interested in building them from source, check the `build instructions on GitHub <https://github.com/openvinotoolkit/openvino/blob/master/docs/dev/build.md>`__ .

.. tab-set::

   .. tab-item:: Linux
      :sync: linux

>       The officially supported Linux build environment is the following:

Reviewer 2: this is a duplication of common system requirements. Should we just refer to a generic page with system requirements?
As far as I remember, @kblaszczak-intel was working on such page

Reviewer 3: If you think the official OV docs system requirements page is a proper substitute for this section, we would gladly remove it. The less superficial content the better.

            * C++ samples: ``~/openvino_cpp_samples_build/<architecture>/Release`` where the <architecture> is the output of ``uname -m``, for example, ``intel64``, ``armhf``, or ``aarch64``.
            
            You can also build the sample applications manually:
            
            .. note::
               
               If you have installed the product as a root user, switch to root mode before you continue: ``sudo -i`` .
      
            1. Navigate to a directory that you have write access to and create a samples build directory. This example uses a directory named ``build``:
            
               .. code-block:: sh
                  
                  mkdir build
               
               .. note:: 
               
                  If you ran the Image Classification verification script during the installation, the C++ samples build directory is created in your home directory: ``~/openvino_cpp_samples_build/``
               
            2. Go to the created directory:
               
               .. code-block:: sh
                  
                  cd build
               

>             3. Run CMake to generate the Make files for release or debug configuration. For example, for C++ samples:
>                
>                - For release configuration:
>             
>                  .. code-block:: sh
>                   
>                     cmake -DCMAKE_BUILD_TYPE=Release <INSTALL_DIR>/samples/cpp

Reviewer: ```suggestion
            3. Run CMake to generate the Make files for release configuration. For example, for C++ samples:
            
                .. code-block:: sh
                  
                   cmake -DCMAKE_BUILD_TYPE=Release <INSTALL_DIR>/samples/cpp
```

               
               - For debug configuration:
            
                 .. code-block:: sh
                    

>                     cmake -DCMAKE_BUILD_TYPE=Debug <INSTALL_DIR>/samples/cpp

Reviewer 2: I think we can remove demonstration of building samples in Debug.
It's not related to samples, but rather to cmake features

            
            4. Run ``make`` to build the samples:
            
               .. code-block:: sh
                  

>                   make

Reviewer 2: `cmake --build . --parallel` instead of `make`

>                   make

Reviewer: ```suggestion
                  cmake --build . --parallel
```

               
            2. Go to the created directory:
            
               .. code-block:: sh
               
                  cd build
            
            3. Run CMake to generate the Make files for release or debug configuration. For example, for C++ samples:
            
               - For release configuration:
            
                 .. code-block:: sh
            
                    cmake -DCMAKE_BUILD_TYPE=Release <INSTALL_DIR>/samples/cpp
               
               - For debug configuration:
            
                 .. code-block:: sh
            
                    cmake -DCMAKE_BUILD_TYPE=Debug <INSTALL_DIR>/samples/cpp
               
            4. Run ``make`` to build the samples:
            
               .. code-block:: sh
               
                  make
            
            For the release configuration, the sample application binaries are in ``<path_to_build_directory>/<architecture>/Release/``; for the debug configuration — in ``<path_to_build_directory>/<architecture>/Debug/``.
      

.. _select-sample:

> Sample Application Setup

Reviewer 2: in general, this section just initializes OpenVINO. And it's already done in the "Run inference on a sample" section

Reviewer: I have applied required changes in the https://github.com/openvinotoolkit/openvino/pull/20957/commits/629ffb13ccb5cf5018dd0385c03a6a064bcb134a commit.

================================

Before running compiled binary files, run the ``setupvars`` script to set all necessary environment variables:

.. tab-set::

   .. tab-item:: Linux
      :sync: linux

      .. code-block:: sh
   
         source <INSTALL_DIR>/setupvars.sh
      
      The OpenVINO environment variables are removed when you close the shell. As an option, you can permanently set the environment variables as follows:

      1. Open the ``.bashrc`` file in ``<user_home_directory>``:
      
         .. code-block:: sh
            

>             vi <user_home_directory>/.bashrc

Reviewer 2: I suppose this part is connected to OpenVINO installation and we already recommended users to perform this step in this document https://docs.openvino.ai/2023.2/openvino_docs_install_guides_installing_openvino_from_archive_linux.html#step-2-configure-the-environment in the **Note** section. So, no needs to duplicate the material

Reviewer: Applied required changes in the https://github.com/openvinotoolkit/openvino/pull/20957/commits/629ffb13ccb5cf5018dd0385c03a6a064bcb134a commit.

                 toolkit, and how to run inference, using provided code samples.

> To use OpenVINO samples, you need to install OpenVINO using one of the distributions containing them. Follow one of the installation guides: 

Reviewer 3: ```suggestion
To use OpenVINO samples, install OpenVINO using one of the following distributions: 
```

* Archive files (recommended) - :doc:`Linux <openvino_docs_install_guides_installing_openvino_from_archive_linux>` | :doc:`Windows <openvino_docs_install_guides_installing_openvino_from_archive_windows>` | :doc:`macOS <openvino_docs_install_guides_installing_openvino_from_archive_macos>`
* :doc:`APT <openvino_docs_install_guides_installing_openvino_apt>` or :doc:`YUM <openvino_docs_install_guides_installing_openvino_yum>` for Linux
* :doc:`Docker image <openvino_docs_install_guides_installing_openvino_docker>`
* `Build from source <https://github.com/openvinotoolkit/openvino/blob/master/docs/dev/build.md>`__

> If you installed OpenVINO Runtime via archive files, sample applications for Python, C++ and C are created in the following directories:

Reviewer 3: ```suggestion
If you install OpenVINO Runtime via archive files, sample applications are created in the following directories:
```

* ``<INSTALL_DIR>/samples/python``
* ``<INSTALL_DIR>/samples/cpp``
* ``<INSTALL_DIR>/samples/c``

.. note::

>    If you install OpenVINO via PyPI, you can still download `the OpenVINO repository <https://github.com/openvinotoolkit/openvino/>`__ and use samples from ``samples/python``.

Reviewer 3: ```suggestion
   If you install OpenVINO without samples, you can still get the python ones directly from `the OpenVINO repository <https://github.com/openvinotoolkit/openvino/>`__.
```
